### PR TITLE
[AIRFLOW-766] Skip conn.commit() when in Auto-commit

### PR DESCRIPTION
**Pull request by pfubar**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:

https://issues.apache.org/jira/browse/AIRFLOW-766

I created brand new merge request since #2002 was having a lot of merge conflict on common files like README.md

Thanks

### CODE REVIEW

**Comment by pfubar** [comment]
@smarden1, @artwr, @jomar83, and @bolkedebruin 
Please review, thank you

File: airflow/utils/db.py [hunk 1]
```diff
@@ -245,6 +245,17 @@ def initdb():
                     ]
                 }
             '''))
+    merge_conn(
```
Comment (bolkedebruin) [inline airflow/utils/db.py]: Why is this added? It won't work for other users than yourself? If it is for testing purposes, why don't you just add it in the setup part of the tests?

**Comment by aarondeutsch** [comment]
Hi, is there a way to review this pull request? Specifically, JDBC autocommit should be changed to conn.jconn.setAutoCommit(autocommit) in jdbc_hook.

@smarden1, @artwr, @jomar83, and @bolkedebruin

**Comment by artwr** [comment]
Trying to rebase and fix https://github.com/apache/incubator-airflow/pull/3257
@pfubar @aarondeutsch
